include/fiction/algorithms/physical_design/graph_oriented_layout_design.hpp

```diff
@@ -0,0 +1,1646 @@
+     */
+    std::optional<Lyt> run() noexcept
+    {
+        // measure run time
+        mockturtle::stopwatch stop{pst.time_total};
+
+        // initialize layout to keep track of current best solution
+        Lyt best_lyt{{}, twoddwave_clocking<Lyt>()};
+
+        // initialize search space graphs
+        initialize();
+
+        // check if a timeout was set
+        bool timeout_set = (timeout != std::numeric_limits<uint64_t>::max());
+
+        // if no timeout was set and high-effort mode is disabled, set timeout to 10s
+        if (!timeout_set && !ps.high_effort_mode)
+        {
+            timeout = 10000u;
+        }
+        bool timeout_limit_reached = false;
+
+#if (PROGRESS_BARS)
+        // initialize a progress bar
+        mockturtle::progress_bar bar{ssg_vec[0].network.size(), "[i] arranging layout: |{0}|"};
+        uint64_t                 max_placed_nodes_tracker = 0u;
+#endif
+
+        // main A* loop
+        while (!timeout_limit_reached)
+        {
+            for (auto& ssg : ssg_vec)
+            {
+                if (ssg.frontier_flag)
+                {
+                    num_evaluated_paths++;
+                    const auto expansion = expand(ssg);
+                    if (expansion.second)
+                    {
+                        best_lyt = *expansion.second;
+                        restore_names(ssg.network, best_lyt);
+
+                        // statistical information
+                        pst.x_size    = best_lyt.x() + 1;
+                        pst.y_size    = best_lyt.y() + 1;
+                        pst.num_gates = best_lyt.num_gates();
+                        pst.num_wires = best_lyt.num_wires();
+
+                        if (ps.return_first)
+                        {
+                            return best_lyt;
+                        }
+                    }
+                    for (const auto& [next, cost] : expansion.first)
+                    {
+                        if (ssg.cost_so_far.find(next) == ssg.cost_so_far.cend() || cost < ssg.cost_so_far[next])
+                        {
+                            ssg.cost_so_far[next] = cost;
+                            double priority       = cost;
+                            ssg.frontier.put(next, priority);
+                        }
+                    }
+                }
+            }
+            for (auto& ssg : ssg_vec)
+            {
+                if (ssg.frontier_flag)
+                {
+                    if (!ssg.frontier.empty())
+                    {
+                        ssg.current_vertex = ssg.frontier.get();
+                    }
+                    else
+                    {
+                        ssg.frontier_flag = false;
+                    }
+
+#if (PROGRESS_BARS)
+                    if (max_placed_nodes > max_placed_nodes_tracker)
+                    {
+                        max_placed_nodes_tracker = max_placed_nodes;
```

Comment:
Is this the intended behavior? I expected the tracker to be slowly increased with every step in the graph that has been evaluated.